geocity/apps/permits/models.py

```diff
@@ -1752,24 +1754,19 @@ class ArchivedPermitRequest(models.Model):
         on_delete=models.CASCADE,
         primary_key=True,
     )
-
-    @property
-    def dirname(self):
-        archived_date = self.archived_date.strftime("%d.%m.%Y.%H.%M.%S")
-        return f"{self.permit_request.id:02d}_{archived_date}-{slugify(self.permit_request.get_works_type_names_list())}"
+    archive = fields.ArchiveDocumentFileField(_("Archive"))
 
     @property
     def path(self):
-        return os.path.join(settings.ARCHIVE_ROOT, self.dirname)
+        return self.archive.path
 
     def delete(self, using=None, keep_parents=False):
-        # delete the archive
-        shutil.rmtree(self.path)
-
-        # delete the permit request linked to the archive
+        # TODO: is this really wanted ?!!
         self.permit_request.delete()
```

Comment:
@monodo Stumbled about this: seems deleting an `ArchivedPermitRequest` cascades to it's `PermitRequest`, which I find very counter-intuitive, and could easily lead to future data-loss. I think we should remove that and put that code in the view that handles it.

Reply:
Indeed. So , you would add a button that batch deletes selected permitrequests and associated files, that's right ? I'll fill an issue for that